packages/iov-bns/src/bnsconnection.posttxs.spec.ts

```diff
@@ -930,7 +930,7 @@ describe("BnsConnection (post txs)", () => {
       assert(contractId, "Contract ID must be set");
       // Contract ID is an incrementing fixed length uint64
       expect(contractId.length).toEqual(8);
-      expect(Uint64.fromBytesBigEndian(contractId).toNumber()).toBeLessThan(800);
+      expect(decodeNumericId(contractId)).toBeLessThan(Number.MAX_SAFE_INTEGER);
```

Comment:
Here you want to convert from the blockchain type to the JavaScript type ASAP. E.g.

```ts
const { result: createContractResult, transaction: firstSearchResultTransaction } = searchResult1[0];
assert(createContractResult, "Contract ID must be set in the result of the create contract transaction");
const contractId = decodeNumericId(createContractResult);
```

Then there is no point in testing anything about `contractId` anymore, since `decodeNumericId` ensures it is in the safe integer range.

Reply:
https://github.com/iov-one/iov-core/pull/1176/commits/545df0dc6ab749131fd0fb2a08bbef3bc5191069 (wrong branch; I need coffee)